master/internal/resourcemanagers/resource_pool.go

```diff
@@ -264,17 +261,27 @@ func (rp *ResourcePool) Receive(ctx *actor.Context) error {
 
 	case GetResourceSummary:
 		reschedule = false
-		ctx.Respond(getResourceSummary(rp.agents))
+		rp.agentStatesCache = rp.fetchAgentStates(ctx)
+		defer func() {
+			rp.agentStatesCache = nil
+		}()
+		ctx.Respond(getResourceSummary(rp.agentStatesCache))
 
 	case aproto.GetRPConfig:
 		reschedule = false
 		ctx.Respond(aproto.GetRPResponse{
-			AgentReconnectWait:   rp.config.AgentReconnectWait,
-			AgentReattachEnabled: rp.config.AgentReattachEnabled,
+			AgentReconnectWait:    rp.config.AgentReconnectWait,
+			AgentReattachEnabled:  rp.config.AgentReattachEnabled,
+			MaxZeroSlotContainers: rp.config.MaxAuxContainersPerAgent,
 		})
 
 	case schedulerTick:
 		if rp.reschedule {
+			rp.agentStatesCache = rp.fetchAgentStates(ctx)
```

Comment:
thoughts on letting agents push updates instead of this?

Reply:
that's a good Q. it'd mean we'd have to sprinkle something like `agent.notifyResourcePool` all around the code to tell it what's up. which feels like it'd be easy to miss. 
I can't think of a way to automate it in go without having explicit getters/setters for all fields on the state.

Reply:
yeah, good point. maybe we can make scheduler tick configurable as an escape hatch..